### PR TITLE
Refactor/2318/migrate node ctx menu 4

### PR DESCRIPTION
**Pull request by shaman-apprentice**
migrate complete node-context-menu

### CODE REVIEW

File: visualization/app/codeCharta/state/effects/nodeContextMenu/nodeContextMenu.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		const nodeContextMenu = TestBed.inject(NodeContextMenuService)
+		const closeSpy = jest.spyOn(nodeContextMenu, "close")
+		nodeContextMenu.open(10, 10)
+
+		document.dispatchEvent(new MouseEvent("click"))
+
+		expect(closeSpy).toHaveBeenCalled()
+	})
+
+	it("should not close when clicking within (its) color picker", () => {
```
Comment (Hall-Ma) [inline visualization/app/codeCharta/state/effects/nodeContextMenu/nodeContextMenu.service.spec.ts:34]: When I tested this test case in the application, the menu still closed on almost every mouse click. In our demo version this does not happen. Could you maybe take a look at it again?

**Review by Hall-Ma — APPROVED** [review]
LGTM!
Please add a changelog entry.

File: visualization/app/codeCharta/state/effects/nodeContextMenu/nodeContextMenu.service.spec.ts (hunk 1 above)
Comment (shaman-apprentice) [reply visualization/app/codeCharta/state/effects/nodeContextMenu/nodeContextMenu.service.spec.ts:34]: As far as I know the following is the only case, where the ctx menu should not close. And this works on my machine :S
image

**Comment by shaman-apprentice** [comment]
> LGTM! Please add a changelog entry.

What are you missing in the existing one? :)

**Comment by Hall-Ma** [comment]
> > LGTM! Please add a changelog entry.
> 
> What are you missing in the existing one? :)

Ups... I thought I hadn't found it😅 Then everything looks good!

**Review by Hall-Ma — APPROVED** [review]

**Pull request merged**